Mathlib/Algebra/IsPrimePow.lean

```diff
@@ -82,15 +83,23 @@ theorem Nat.Prime.isPrimePow {p : ℕ} (hp : p.Prime) : IsPrimePow p :=
 #align nat.prime.is_prime_pow Nat.Prime.isPrimePow
 
 theorem isPrimePow_nat_iff_bounded (n : ℕ) :
-    IsPrimePow n ↔ ∃ p : ℕ, p ≤ n ∧ ∃ k : ℕ, k ≤ n ∧ p.Prime ∧ 0 < k ∧ p ^ k = n := by
+    IsPrimePow n ↔ ∃ p ≤ n, ∃ k ≤ n.log2, p.Prime ∧ 0 < k ∧ p ^ k = n := by
   rw [isPrimePow_nat_iff]
-  refine' Iff.symm ⟨fun ⟨p, _, k, _, hp, hk, hn⟩ => ⟨p, k, hp, hk, hn⟩, _⟩
+  refine Iff.symm ⟨fun ⟨p, _, k, _, hp, hk, hn⟩ => ⟨p, k, hp, hk, hn⟩, ?_⟩
   rintro ⟨p, k, hp, hk, rfl⟩
-  refine' ⟨p, _, k, (Nat.lt_pow_self hp.one_lt _).le, hp, hk, rfl⟩
-  conv => { lhs; rw [← (pow_one p)] }
-  exact pow_le_pow_right hp.one_lt.le hk
+  refine ⟨p, ?_, k, Nat.le_log2_prime_pow hp, hp, hk, rfl⟩
+  exact Nat.le_self_pow (Nat.pos_iff_ne_zero.mp hk) _
 #align is_prime_pow_nat_iff_bounded isPrimePow_nat_iff_bounded
 
+theorem isPrimePow_nat_iff_bounded' (n : ℕ) :
```

Comment:
Have you copied the same theorem twice? I think once is enough :-)

Reply:
Thinking about this result slightly more: I'm not sure if it's *that* mathematically interesting - but I don't mind "easy" strengthenings of existing results either.

Reply:
> Have you copied the same theorem twice? I think once is enough :-)

Thanks for your interest! These two theorems are very close but different in that the first one has the bound of n.log2 (the best bound without dependence to p) and the second one has p.log n (the best bound that depends on p). (The second one has one more line in the proof for this reason)

Reply:
> Thinking about this result slightly more: I'm not sure if it's _that_ mathematically interesting - but I don't mind "easy" strengthenings of existing results either.

Yeah, I made this PR because while I was looking around Mathlib, I found a bound that didn't seem optimal and thought it would be a easy win.

But then, I wasn't sure whether the new bound be allowed to depend on p or not so I tried both cases and soon after, exams swept me away.